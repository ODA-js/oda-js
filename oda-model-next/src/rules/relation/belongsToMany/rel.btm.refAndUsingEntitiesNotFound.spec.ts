import 'jest';
import Rule from './refAndUsingEntitiesNotFound';

describe('rule', () => {
  let rule: Rule;
  beforeAll(() => {
    rule = new Rule();
  });

  it('throw', () => {
    const updateWith = jest.fn();
    const result = rule.validate({
      field: {
        updateWith,
      },
      relation: {
        modelType: 'relation',
        verb: 'BelongsToMany',
        ref: {
          entity: true,
        },
        using: {
          entity: true,
        },
      },
      package: {
        items: {
          get: jest
            .fn()
            .mockReturnValueOnce({
              modelType: 'not entity',
            })
            .mockReturnValueOnce({
              modelType: 'not entity',
            }),
        },
      },
    } as any);
    expect(updateWith).toBeCalledWith({ relation: null });
    expect(result).toMatchSnapshot();
  });

  it('not throw 1', () => {
    const updateWith = jest.fn();
    const result = rule.validate({
      field: {
        updateWith,
      },
      relation: {
        modelType: 'relation',
        verb: 'BelongsToMany',
        ref: {
          entity: true,
        },
        using: {
          entity: true,
        },
      },
      package: {
        items: {
          get: jest
            .fn()
            .mockReturnValueOnce({
              modelType: 'entity',
            })
            .mockReturnValueOnce({
              modelType: 'entity',
            }),
        },
      },
    } as any);
    expect(updateWith).not.toBeCalledWith();
    expect(result).toMatchSnapshot();
  });
});
