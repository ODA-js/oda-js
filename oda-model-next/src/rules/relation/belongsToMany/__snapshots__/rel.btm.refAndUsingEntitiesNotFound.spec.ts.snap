// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule not throw 1 1`] = `Array []`;

exports[`rule throw 1`] = `
Array [
  Object {
    "message": "referenced and using entities not found. remove relation",
    "result": "fixable",
  },
]
`;
